src/components/ContentWidgets/Gallery/GalleryEditor.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { WidgetEditorProps } from '../../ContentMapping/ContentMapping';
+import './Gallery.css';
+
+export const GalleryEditor: React.FC<WidgetEditorProps> = ({
+    editedContent,
+    setEditedContentOnChange
+}) => {
+    const galleryContent = editedContent.gallery_content || [];
+
+    const updateGallery = (link: string, index: number) => {
```

Comment:
These two functions can be put outside of the component, and then you pass in the dependencies it needs (galleryContent). It's a better design for testing functions as well as creating instances of the function.

Reply:
How does setEditedContentOnChange() work if the function is outside of the widget?

Reply:
You would probably have it be passed through the function as a parameter.